lib/ash/api/api.ex

```diff
@@ -1194,6 +1195,17 @@ defmodule Ash.Api do
     end
   end
 
+  defp pagination_check(action, resource, opts) do
+    case Keyword.has_key?(opts, :page) && Map.get(action, :pagination) do
+      true ->
+        {:ok, action}
+
+      false ->
+        {:error,
```

Comment:
This is a great start! 

Two things:
1. If you don't mind, can you create a new `invalid` error class module for it? Then, you'd return `Ash.Error.to_error_class(YourNewModule.exception(resource: resource, action: action))`.

2. include the action in the message. Pagination is enabled on an action-by-action basis.

Great work so far!

Reply:
Thanks for the feedback, Zach. I was working on exactly this when writing the test but I was stuck. I guess it was because I was adding it to the forbidden class? Will try with the invalid class and push here asap.

Does the error message look good?

Reply:
It looks good, but should be updated to something like this:
```elixir
Pagination is not enabled on the #{inspect action} action on #{inspect resource}.
```

Reply:
The first sentence, I mean.